src/app/core/filter/filters/filters.ts

```diff
@@ -153,113 +83,37 @@ export class SelectableFilter<T extends Entity> extends Filter<T> {
     public label: string = name,
   ) {
     super(name, label);
-    this.selectedOption = this.options[0]?.key;
+    this.selectedOptionsKeys = this.options.map(
+      (value: FilterSelectionOption<T>) => value.key,
+    );
   }
 
-  /** default filter will keep all items in the result */
-  defaultFilter = {};
-
   /**
    * Get the full filter option by its key.
    * @param key The identifier of the requested option
    */
-  getOption(key: string): FilterSelectionOption<T> {
-    return this.options.find((option) => option.key === key);
+  getOption(key: string): FilterSelectionOption<T> | undefined {
+    return this.options.find((option: FilterSelectionOption<T>): boolean => {
```

Comment:
You don't really need to always set the types. They are inferred automatically. Keeps to code a bit more precise.